cookbook/case_studies/house_price_prediction/house_price_predictor.py

```diff
@@ -0,0 +1,232 @@
+        ),
+        pd.DataFrame(
+            _val,
+            columns=COLUMNS,
+        ),
+        pd.DataFrame(
+            _test,
+            columns=COLUMNS,
+        ),
+    )
+
+
+# %%
+# Step 4: Task -- Generating & Splitting the Data
+# -----------------------------------------------
+#
+# Train, validation, and test datasets are DataFrames.
+@task(cache=True, cache_version="0.1", limits=Resources(mem="600Mi"))
+def generate_and_split_data(
+    loc: str, number_of_houses: int, seed: int
+) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    _houses = gen_houses(number_of_houses)
+    return split_data(_houses, seed, split=SPLIT_RATIOS)
+
+
+# %%
+# Step 5: Task -- Training the XGBoost Model
+# ------------------------------------------
+#
+# Serialize the XGBoost model using joblib and store the model in a dat file.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def fit(
+    loc: str, train: pd.DataFrame, val: pd.DataFrame
+) -> FlyteFile[typing.TypeVar("joblib.dat")]:
+
+    # Fetch the input and output data from train dataset
+    x = train[train.columns[1:]]
+    y = train[train.columns[0]]
+
+    # Fetch the input and output data from validation dataset
+    eval_x = val[val.columns[1:]]
+    eval_y = val[val.columns[0]]
+
+    m = XGBRegressor()
+    m.fit(x, y, eval_set=[(eval_x, eval_y)])
+
+    fname = "model-" + loc + ".joblib.dat"
+    joblib.dump(m, fname)
+    return fname
+
+
+# %%
+# Step 6: Task -- Generating the Predictions
+# ------------------------------------------
+#
+# Unserialize the XGBoost model using joblib and generate the predictions.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def predict(
+    test: pd.DataFrame,
+    model_ser: FlyteFile[typing.TypeVar("joblib.dat")],
+) -> typing.List[float]:
+
+    # Load model
+    model = joblib.load(model_ser)
+
+    # Load test data
+    x_df = test[test.columns[1:]]
+
+    # Generate predictions
+    y_pred = model.predict(x_df).tolist()
+
+    return y_pred
+
+
+# %%
+# Step 7: Workflow -- Defining the Workflow
+# -----------------------------------------
+#
+# #. Generate and split the data
+# #. Fit the XGBoost model
+# #. Generate predictions
+@workflow
+def house_price_predictor_trainer(
+    seed: int = 7, number_of_houses: int = NUM_HOUSES_PER_LOCATION
+) -> typing.List[float]:
+
+    # Generate and split the data
+    train, val, test = generate_and_split_data(
+        loc="NewYork_NY", number_of_houses=number_of_houses, seed=seed
+    )
+
+    # Fit the XGBoost model
+    model = fit(loc="NewYork_NY", train=train, val=val)
+
+    # Generate predictions
+    predictions = predict(model_ser=model, test=test)
+
+    return predictions
+
+
+# %%
+# You can call the ``house_price_predictor_trainer`` workflow to print the predictions. However, our goal is to expand this to multiple regions. Thus, the tasks and helper functions defined here are called in the other Python script.
```

Comment:
do you want to have a __main__ section here to show how to run this example

Reply:
Done.